src/client/proxy/http.cc

```diff
@@ -313,7 +313,11 @@ bool HTTPMessage::CreateHTTPRequest() {
     // m_ErrorResponse is set in ExtractIncomingRequest
     return false;
   }
-  HandleJumpService();
+
+  // Valid HTTP request, but not a valid jump service request
+  if (!HandleJumpService())
+    LOG(debug) << "HTTPMessage: not a valid jump service request";
```

Comment:
If the jump request is invalid, simply proceed to the address?! But we don't have the address in our addressbook (assuming at this point we've confirmed that it does not exist in the addressbook)!  That is another issue to look out for: confirm that we don't already have the address in the addressbook before inserting. See https://github.com/monero-project/kovri/issues/814.

Reply:
Currently, the false return means that the request is a valid HTTP request, but not a valid jump service request. 

There isn't a way to distinguish whether it was:
- intended to be a jump service request (should return a bad_request response)
- a non-jump service request (should continue processing)

So far, I was thinking of moving `HandleJumpService`'s internal regex check into a separate helper function, and performing the check in `CreateHTTPRequest` instead/as well. 

The way it's implemented now, the refactor might be a bit hacky. 

I was thinking short-to-mid-term about entirely rewriting HTTP proxy with `Boost.Beast` or `cpp-netlib`, instead of the custom HTTP parsing code. Do you have advice on doing so, or if it is even a good idea?

Reply:
>Currently, the false return means that the request is a valid HTTP request, but not a valid jump service request. At least, that's how I was thinking about it.

But you need to follow the flow. See #814. If what you want is not yet implemented, be sure to implement *before* you PR.

>So far, I was thinking of moving HandleJumpService's internal regex check into a separate helper function, and performing the check in CreateHTTPRequest instead/as well.

The entire proxy needs a rewrite. Do whatever you need to do.

>Do you have advice on doing so, or if it is even a good idea?

1. https://github.com/monero-project/kovri/issues/721
2. Beast doesn't have proxy support out of the box so you're looking at a chunk of work regardless of lib usage
3. Boost.beast is now in 1.66 but default ubuntu boost is *not* 1.66. As such, we can't depend on a lib which has yet to be readily available. This not only applies to linux but all of our platforms on build.getmonero.org

Reply:
> But you need to follow the flow. See #814. If what you want is not yet implemented, be sure to implement before you PR.

This was an oversight on my part, patched now with a fix (at least where parsing is concerned). I'll work on the "save-to-address-book" workflow in another PR. Updated TODO in latest patch.

Will also be working on a PR for #814 next.

Regarding Boost.beast, I'll refactor using cpp-netlib where necessary for now.
Will work on a PR for #721 once Boost.beast is supported on all buildbot platforms.

Reply:
>I'll refactor using cpp-netlib where necessary for now.

Do NOT. I repeat, DO NOT. Any further cpp-netlib work will be NACK'd. #721 means NOT adding MORE cpp-netlib.